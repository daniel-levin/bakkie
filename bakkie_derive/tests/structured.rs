#[bakkie::structured]
enum B {
    Something { field: String },

    Nothing,
}

#[bakkie::structured]
struct S {
    a: String,
    b: B,
}

#[allow(dead_code)]
fn can_json() {
    let _: S = serde_json::from_str("").unwrap();
}
