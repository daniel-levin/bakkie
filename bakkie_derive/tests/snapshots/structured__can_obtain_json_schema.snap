---
source: bakkie_derive/tests/structured.rs
expression: schema
---
{
  "$defs": {
    "B": {
      "oneOf": [
        {
          "enum": [
            "Nothing"
          ],
          "type": "string"
        },
        {
          "additionalProperties": false,
          "properties": {
            "Something": {
              "properties": {
                "field": {
                  "type": "string"
                }
              },
              "required": [
                "field"
              ],
              "type": "object"
            }
          },
          "required": [
            "Something"
          ],
          "type": "object"
        }
      ]
    }
  },
  "$schema": "https://json-schema.org/draft/2020-12/schema",
  "properties": {
    "a": {
      "type": "string"
    },
    "b": {
      "$ref": "#/$defs/B"
    }
  },
  "required": [
    "a",
    "b"
  ],
  "title": "S",
  "type": "object"
}
